File: src/main/java/com/zaze/server/service/ShowcaseService.java
package com.zaze.server.service;

import java.util.List;

import com.zaze.server.model.Showcase;
import org.springframework.data.domain.Pageable;

public interface ShowcaseService {

    void saveShowcase(Showcase showcase);

    void deleteShowcases(long[] ids);

    List<Showcase> getShowcaseList();

    List<Showcase> findByTagsLike(Pageable pageable, String tags);
}

File: feature/showcase/bin/main/com/zaze/server/feature/showcase/service/ShowcaseService.kt
package com.zaze.server.feature.showcase.service

import com.zaze.server.feature.showcase.vo.ShowcaseVo
import org.springframework.data.domain.Pageable

// service 可以定义一个dto 使用，这里直接使用vo
interface ShowcaseService {
    fun saveShowcase(showcase: ShowcaseVo): ShowcaseVo
    fun deleteShowcases(ids: List<Long>)
    fun findByTagsLike(pageable: Pageable, tags: String): List<ShowcaseVo>
    fun getShowcaseList(): List<ShowcaseVo>
}